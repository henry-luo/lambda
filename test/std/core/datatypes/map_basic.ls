// Test: Map/Dictionary Basic Operations
// Category: core/datatypes
// Type: positive
// Expected: 1

// Map literals
let empty_map = {}
empty_map
let person = {
person
    name: "Alice",
    age: 30,
    active: true,
    scores: [95, 87, 92]
}
let name = person.name      
name
let age = person["age"]     
age
let missing = person.email  
missing
person.city = "New York"
person["country"] = "USA"
person.age += 1  // 31
let keys = Object.keys(person)    
keys
let values = Object.values(person) 
values
let entries = Object.entries(person) 
entries
let hasName = "name" in person  
hasName
let hasEmail = "email" in person 
hasEmail
let company = {
company
    name: "Tech Corp",
    employees: [
        {name: "Alice", role: "Developer"},
        {name: "Bob", role: "Designer"}
    ],
    address: {
        street: "123 Tech St",
        city: "San Francisco"
    }
}
let first_employee = company.employees[0].name  
first_employee
let city = company.address.city  
city
1
