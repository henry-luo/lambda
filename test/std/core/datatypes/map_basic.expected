09:15:42 [NOTICE] Start transpiling test/std/core/datatypes/map_basic.ls...
09:15:42 [ERROR] Syntax tree has errors.
09:15:42 [ERROR] Syntax error at Ln 21, Col 1 - 21, Col 14: ERROR
09:15:42 [ERROR] ERROR node at Ln 21, Col 1 - 21, Col 14
09:15:42 [ERROR] Syntax error at Ln 22, Col 1 - 22, Col 20: ERROR
09:15:42 [ERROR] ERROR node at Ln 22, Col 1 - 22, Col 20
09:15:42 [ERROR] Syntax error at Ln 23, Col 13 - 23, Col 14: ERROR
09:15:42 [ERROR] ERROR node at Ln 23, Col 13 - 23, Col 14
09:15:42 [ERROR] Error: Failed to compile the function.
##### Script 'test/std/core/datatypes/map_basic.ls' executed: #####
null
